src/main/editDelete/preProcess.ts

```diff
@@ -50,6 +31,74 @@ const injectAttributes: MapCallback<PartialWord, Word> = (word, i) => ({
   fileName: 'PLACEHOLDER FILENAME',
 });
 
+const constructWord = (
+  word: string,
+  startTime: number,
+  duration: number,
+  outputStartTime: number,
+  key: string
+) => {
+  const newWord: Word = {
+    word,
+    startTime: roundToMs(startTime),
+    duration: roundToMs(duration),
+    outputStartTime: roundToMs(outputStartTime),
+    deleted: false,
+    key,
+    fileName: 'PLACEHOLDER FILENAME',
+  };
+
+  return newWord;
+};
+
+/**
+ * Adds spaces between words which represent the silence between each word
+ * @param result The output being constructed
+ * @param word The current element of words
+ * @param index The index of word in words
+ * @param words The list of words being reduced
+ * @returns The updated transcript with a silence after word
+ */
+let TOTAL_DURATION = 0;
+const addSpaces = (
+  result: Word[],
+  word: Word,
+  index: number,
+  words: Word[]
+) => {
+  const spaceChar = ' ';
+
+  // is the first word
+  if (index === 0) {
+    result.push(
+      constructWord(spaceChar, 0, words[index].startTime, 0, index.toString())
+    );
+  }
+
+  const isLastWord = index === words.length - 1;
+  let silenceDuration;
+  if (isLastWord) {
+    silenceDuration = TOTAL_DURATION - word.startTime - word.duration;
+  } else {
+    silenceDuration =
+      words[index + 1].startTime - word.startTime - word.duration;
+  }
+
+  word.key = (index * 2 + 1).toString();
```

Comment:
Add comments explaining what this index is for and why it's the n*2+1'th, otherwise they're just magic numbers

Reply:
Good point, added some comments here: https://github.com/patrickbrett/mlvet/pull/54/commits/ff252792d7273be27cb6c53d3faa103ad833af00